test/backbone.blazer.base-router.spec.js

```diff
@@ -0,0 +1,640 @@
+
+  describe('when routing to a matched Regex route with pushState', function() {
+    beforeEach(function() {
+      Backbone.history.location = new this.Location('http://example.com/example');
+      this.routeObj = {};
+      this.router.route(/^example$/, this.routeObj);
+      Backbone.history.start({pushState: true});
+      this.router.navigate('example', {trigger: true});
+    });
+
+    it('should not include originalRoute in the options', function() {
+      expect(this.routeData).to.not.include.keys('originalRoute');
+    });
+
+    it('should trigger onNavigate', function() {
+      expect(this.router.onNavigate).to.have.been.calledOnce;
+    });
+
+    it('should pass the object that was linked with the route', function() {
+      expect(this.routeData.linked).to.deep.equal(this.routeObj);
+    });
+
+    it('should pass the Router instance', function() {
+      expect(this.routeData.router).to.equal(this.router);
+    });
+
+    it('should pass the current URI fragment', function() {
+      expect(this.routeData.uriFragment).to.equal('example');
+    });
+
+    it('should pass a Regular expression as the route', function() {
+      expect(this.routeData.route).to.be.instanceOf(RegExp);
+    });
+
+    it('should pass an empty object for query params', function() {
+      expect(this.routeData.query).to.deep.equal({});
+    });
+
+    it('should pass empty named params', function() {
+      expect(this.routeData.params).to.deep.equal({});
+    });
+  });
+
+  describe('when routing to a matched Regex route without pushState', function() {
+    beforeEach(function() {
+      Backbone.history.location = new this.Location('http://example.com#example');
+      this.routeObj = {};
+      this.router.route(/^example$/, this.routeObj);
+      Backbone.history.start({pushState: false});
+      this.router.navigate('example', {trigger: true});
+    });
+
+    it('should not include originalRoute in the options', function() {
+      expect(this.routeData).to.not.include.keys('originalRoute');
+    });
+
+    it('should trigger onNavigate', function() {
+      expect(this.router.onNavigate).to.have.been.calledOnce;
+    });
+
+    it('should pass the object that was linked with the route', function() {
+      expect(this.routeData.linked).to.deep.equal(this.routeObj);
+    });
+
+    it('should pass the Router instance', function() {
+      expect(this.routeData.router).to.equal(this.router);
+    });
+
+    it('should pass the current URI fragment', function() {
+      expect(this.routeData.uriFragment).to.equal('example');
+    });
+
+    it('should pass a Regular expression as the route', function() {
+      expect(this.routeData.route).to.be.instanceOf(RegExp);
+    });
+
+    it('should pass an empty object for query params', function() {
+      expect(this.routeData.query).to.deep.equal({});
+    });
+
+    it('should pass empty named params', function() {
+      expect(this.routeData.params).to.deep.equal({});
+    });
+  });
+});
```

Comment:
i wanted to make sure that we didn't break the contract that base-router provided. i think now that it's internalized as a dependency in this PR that maybe it's even more valuable to have these tests, no?